Store: Fix ESLint errors in WCS, tests, lib, and components directories

justinshreve opened the pull request:
See #24504.

This fixes the eslint issues in the following files:

* client/extensions/woocommerce/components/bulk-select/index.js (2 err, 0 warn)
* client/extensions/woocommerce/components/form-click-to-edit-input/index.js (2 err, 0 warn)
* client/extensions/woocommerce/components/form-location-select/countries.js (2 err, 0 warn)
* client/extensions/woocommerce/components/location-flag/index.js (1 err, 0 warn)
* client/extensions/woocommerce/lib/generate-variations/test/index.js (1 err, 0 warn)
* client/extensions/woocommerce/state/data-layer/ui/products/index.js (3 err, 0 warn)
* client/extensions/woocommerce/state/sites/payment-methods/test/reducer.js (1 err, 0 warn)
* client/extensions/woocommerce/state/sites/review-replies/test/handlers.js (1 err, 0 warn)
* client/extensions/woocommerce/state/sites/reviews/test/handlers.js (1 err, 0 warn)
* client/extensions/woocommerce/state/ui/payments/methods/test/selectors.js (1 err, 0 warn)
* client/extensions/woocommerce/state/ui/shipping/zones/locations/test/selectors.js (1 err, 0 warn)
* client/extensions/woocommerce/state/ui/shipping/zones/methods/selectors.js (2 err, 0 warn)
* client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js (1 err, 0 warn)
* client/extensions/woocommerce/woocommerce-services/views/packages/edit-package.js (1 err, 0 warn)
* client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-item.js (3 err, 0 warn)
* client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/address-step/suggestion.js (3 err, 0 warn)
* client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/list.js (2 err, 0 warn)

Some of these are around duplicate test descriptions, indentation, and buttons vs links. No functionality should be changed so testing can just be spot checking.

To Test:
* Run tests `npm run test-client client/extensions/woocommerce`
* Click around store. Specifically, I've touched the following pages (via components or selectors):
    * SKU Edit Component
    * Address Dialog (Country/State Selection)
    * Shipping zone & method dialogs
    * Product reviews & review replies
    * Label & package dialogs and flow

ryelle commented on client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/list.js:
```diff
@@ -42,12 +43,13 @@ const PackageList = ( props ) => {
 			<div className="packages-step__list-item" key={ pckgId }>
-				<div
+				<Button
+					borderless
```
Now that this is a button, the contrast is a bit light: 

<img width="263" alt="screen shot 2018-05-01 at 1 15 37 pm" src="https://user-images.githubusercontent.com/541093/39483947-0a086000-4d42-11e8-99d6-abc6ae12646d.png">

Can you use the `$gray-text` color for this?

justinshreve commented:
@ryelle Thanks for the review.

I updated the text color for the packages dialog. I also added a TODO to make that a proper selector which I can handle in a follow-up PR.

For the indent/linter one -- I've tried formatting that line a few different ways. Prettier keeps updating it on commit to:

```
		const defaultValues = startsWith( method.methodType, 'wc_services' )
			? getDefaultSettingsValues(
					getShippingMethodSchema( state, method.methodType, siteId ).formSchema
				)
			: {};
```

which is slightly different from what you pasted above. 

The linter will complain "Expected indentation of 4 tabs but found 5" without disabling the line there. I've at least updated the comment to be a bit clearer.

blowery commented:
@justinshreve I wonder if #24551 would help there at all.. /cc @jsnajdr

blowery commented on client/extensions/woocommerce/components/form-location-select/countries.js:
```diff
@@ -98,6 +98,22 @@ class FormCountrySelectFromApi extends Component {
+// https://github.com/Automattic/wp-calypso/pull/24571#discussion_r185268996
+const getContinentsWithCountries = ( state, continents, siteId ) => {
+	const locationsList = [];
+	continents.forEach( continent => {
```
this could be a `map` instead a `forEach` with pushes

justinshreve replied:
Thanks! In this PR I'm just going to move it out of `connect`, but I plan on making this a proper selector + add some tests in a follow-up, so I can do some code cleanup there.

jsnajdr commented:
> I wonder if #24551 would help there at all.

The indenting of ternaries isn't a bug, it's a difference of opinion between Prettier and ESLint. The ESLint `indent` rule doesn't like what Prettier (intentionally) does, that's all. Prettier upgrade won't change that.

blowery commented:
@jsnajdr https://github.com/Automattic/wp-calypso/pull/24635 should resolve that difference of opinion. :)

ryelle commented:
@justinshreve apparently I had a setting on in my editor that ran `eslint --fix` after formatting, so when I used Atom to format the file, it did so with the eslint indentation... but when I ran prettier from CLI it indents like you said. Sorry about any confusion from my comment 😣
